File: src/core/Dataset.cpp
/* GCompris - Dataset.cpp
 *
 * SPDX-FileCopyrightText: 2020 Johnny Jazeix <jazeix@gmail.com>
 *
 * Authors:
 *   Johnny Jazeix <jazeix@gmail.com>
 *
 *   SPDX-License-Identifier: GPL-3.0-or-later
 */
#include "Dataset.h"

Dataset::Dataset(QObject *parent) :
    QObject(parent),
    m_objective(""),
    m_difficulty(0),
    m_enabled(true)
{
}

QString Dataset::objective() const
{
    return m_objective;
}

void Dataset::setObjective(const QString &objective)
{
    m_objective = objective;
    Q_EMIT objectiveChanged();
}

quint32 Dataset::difficulty() const
{
    return m_difficulty;
}
void Dataset::setDifficulty(const quint32 &difficulty)
{
    m_difficulty = difficulty;
    Q_EMIT difficultyChanged();
}

QVariant Dataset::data() const
{
    return m_data;
}
void Dataset::setData(const QVariant &data)
{
    m_data = data;
    Q_EMIT dataChanged();
}

bool Dataset::enabled() const
{
    return m_enabled;
}
void Dataset::setEnabled(const bool &enabled)
{
    m_enabled = enabled;
    Q_EMIT enabledChanged();
}

#include "moc_Dataset.cpp"

File: tools/Dataset.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

from PyQt5.QtCore import pyqtProperty, QObject

class Dataset(QObject):
    def __init__(self, parent=None):
        super().__init__(parent)

        # Initialise the value of the properties.
        self._objective = ''
        self._difficulty = 0
        self._data = ''
        self._enabled = True

    @pyqtProperty('QString')
    def objective(self):
        return self._objective

    @objective.setter
    def objective(self, objective):
        self._objective = objective

    @pyqtProperty(int)
    def difficulty(self):
        return self._difficulty

    @difficulty.setter
    def difficulty(self, difficulty):
        self._difficulty = difficulty

    @pyqtProperty('QVariant')
    def data(self):
        return self._data

    @data.setter
    def data(self, data):
        self._data = data

    @pyqtProperty(bool)
    def enabled(self):
        return self._enabled

    @enabled.setter
    def enabled(self, enabled):
        self._enabled = enabled
